amf-shapes/shared/src/main/scala/amf/shapes/internal/domain/resolution/shape_normalization/ShapeCanonizer.scala

```diff
@@ -239,19 +235,48 @@ sealed case class ShapeCanonizer()(implicit val context: NormalizationContext) e
     }
   }
 
-  def isSimpleInheritance(shape: Shape): Boolean = shape match {
-    case any: AnyShape if any.inherits.size == 1 =>
-      val superType = any.inherits.head
-      val ignoredFields =
-        Seq(ShapeModel.Inherits, AnyShapeModel.Examples, AnyShapeModel.Name)
-      val effectiveFields = shape.fields.fields().filterNot(f => ignoredFields.contains(f.field))
-      validFields(effectiveFields, superType)
-    case _ => false
+  private def isSimpleInheritance(shape: Shape, superTypes: Seq[Shape] = Seq()): Boolean = {
+    val singleSuperType = superTypes match {
+      case Seq(superType) => Some(superType)
+      case _              => None
+    }
+
+    shape match {
```

Comment:
Probably better to pattern match against a tuple ? (Shape, Option[Shape]) ? Or leave the Seq